src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -194,12 +194,25 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
                 this.logger.LogTrace("[RETURN_ATTEMPTED_HC_FAILED]");
                 return attempted;
             }
+            
```

Comment:
Ivan can you please move this out to its own method and write a unit test for it? The `SelectPreferredPeers` method is already getting too bloated ;)

Reply:
I didn't find a nice way to move this piece out into a method without breaking the consistency. 
After creating the extension method this piece became smaller so I think it's alright to keep it inside the `SelectPreferredPeers`